src/main/java/com/pusher/client/channel/impl/PresenceChannelImpl.java

```diff
@@ -104,20 +154,44 @@ public void bind(final String eventName, final SubscriptionEventListener listene
                     "Only instances of PresenceChannelEventListener can be bound to a presence channel");
```

Comment:
This (pre-existing) check here really makes me feel like the current API is broken :(